cpp/src/arrow/util/logging.cc

```diff
@@ -63,33 +64,33 @@ class CerrLog {
  public:
   explicit CerrLog(ArrowLogLevel severity) : severity_(severity), has_logged_(false) {}
 
-  virtual ~CerrLog() {
+  virtual ~CerrLog() noexcept(false) {
     if (has_logged_) {
-      std::cerr << std::endl;
+      stream << std::endl;
     }
     if (severity_ == ArrowLogLevel::ARROW_FATAL) {
-      PrintBackTrace();
-      std::abort();
+      throw std::runtime_error(stream.str());
```

Comment:
It's more or less ok to throw exception here, because this class is used as temporary object in one line logging. Like:
`Log(...) << ...`, so we won't get abort because of throwing exception while stack unwinding.